File: app/src/main/java/com/bsas/androiddevs/manejoerrores/common/exception/ExceptionLogger.java
package com.bsas.androiddevs.manejoerrores.common.exception;

import com.crashlytics.android.Crashlytics;

public class ExceptionLogger {

    private static ExceptionLogger _selfInstance;

    private ExceptionLogger() {
    }

    public static void error(Throwable throwable) {
        _selfInstance.logError(throwable);
    }

    private void logError(Throwable throwable) {
        throwable.printStackTrace();
        Crashlytics.logException(throwable);
    }
}

File: app/src/main/java/com/bsas/androiddevs/manejoerrores/common/logging/ExceptionLogger.kt
package com.bsas.androiddevs.manejoerrores.common.logging

import com.crashlytics.android.Crashlytics

class ExceptionLogger {

    companion object {

        private val exceptionLogger = ExceptionLogger()

        fun error(throwable: Throwable) {
            exceptionLogger.logError(throwable)
        }
    }

    fun logError(throwable: Throwable) {
        throwable.printStackTrace()
        Crashlytics.logException(throwable)
    }

}